src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,311 @@
+                                nullptr, nullptr, ASR::storage_typeType::Default, ASRUtils::TYPE(ASR::make_CPtr_t(al, x.base.base.loc)),
+                                nullptr, ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+                            fn_symtab->add_symbol(s2c(al, "x"), arg);
+                            args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, arg)));
+
+                            Vec<ASR::stmt_t*> body;
+                            body.reserve(al, 1);
+
+                            Vec<char*> dep;
+                            dep.reserve(al, 1);
+
+                            ASR::asr_t* new_subrout = ASRUtils::make_Function_t_util(al, x.base.base.loc,
+                                fn_symtab, s2c(al, new_name), dep.p, dep.n, args.p, args.n, body.p, body.n,
+                                nullptr, ASR::abiType::BindC, ASR::accessType::Public,
+                                ASR::deftypeType::Interface, s2c(al, new_name), false, false, false,
+                                false, false, nullptr, 0, false, false, false, s2c(al, header));
+                            ASR::symbol_t* new_symbol = ASR::down_cast<ASR::symbol_t>(new_subrout);
+                            module_scope->add_symbol(s2c(al, new_name), new_symbol);
+                        }
+
+                        // Extract the symbol from the target (Var)
+                        ASR::symbol_t* var_sym = ASR::down_cast<ASR::Var_t>(x.m_target)->m_v;
+                        ASR::expr_t* target = ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, var_sym));
+
+                        // Create the function call statement for basic_const_pi
+                        ASR::symbol_t* basic_const_pi_sym = module_scope->get_symbol(new_name);
+                        Vec<ASR::call_arg_t> call_args;
+                        call_args.reserve(al, 1);
+                        ASR::call_arg_t call_arg;
+                        call_arg.loc = x.base.base.loc;
+                        call_arg.m_value = target;
+                        call_args.push_back(al, call_arg);
+
+                        ASR::stmt_t* stmt = ASRUtils::STMT(ASR::make_SubroutineCall_t(al, x.base.base.loc, basic_const_pi_sym,
+                            basic_const_pi_sym, call_args.p, call_args.n, nullptr));
+                        pass_result.push_back(al, stmt);
+                        break;
+                    }
+                    default: {
+                        // TODO
+                    }
+                }
+            }
+        }
+    }
+
+    void visit_Print(const ASR::Print_t &x) {
+        std::vector<ASR::expr_t*> print_tmp;
+        SymbolTable* module_scope = current_scope->parent;
+        for (size_t i=0; i<x.n_values; i++) {
+            ASR::expr_t* value = x.m_values[i];
+
+            if (ASR::is_a<ASR::Var_t>(*value) && ASR::is_a<ASR::CPtr_t>(*ASRUtils::expr_type(value))) {
```

Comment:
I might just need some help with this check here . What I was intending to do is catch any `Var` and check if it's of type `CPtr` (now that we've replcaed `S` with `CPtr`) and for those cases use the logic next.
```
                                    (Print
                                        ()
                                        [(Var 3 x)]
                                        ()
                                        ()
                                    )
```

But then we could simply have something like 
```
x: CPtr = empty_c_void_p()
print(x)
```
And we wouldn't like to apply the logic for these cases . Can I somehow make sure we only use this for symbolic cases ?

Reply:
Exactly!